.github/actions/rdp/action.yaml

```diff
@@ -0,0 +1,36 @@
+name: Azure RDP setup
+description: 'Connect via RDP in order to log in as session other than 0.'
+
+inputs:
+  az_vm_hostname:
+    description: Hostname for "xfreerdp /v:az_vm_hostname"
+    required: true
+  az_vm_username:
+    description: Username for "xfreerdp /u:az_vm_username"
+    required: true
+  az_vm_password:
+    description: Password for "xfreerdp /p:az_vm_password"
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    - name: rdp
+      shell: bash
+      run: |
+        set -o pipefail
+        sudo apt-get install -y xvfb freerdp2-x11
+        # detected empirically and confirmed here: https://github.com/FreeRDP/FreeRDP/blob/68ad8d5a1c8d52828095c2293acea1c76fa7fcb5/client/X11/xfreerdp.h#L350
+        vmOfflineErrorCode=141 
+        lastExitCode=$vmOfflineErrorCode
+        maxAttemptsCount=10;
+        while [[ $lastExitCode == $vmOfflineErrorCode && $maxAttemptsCount != 0 ]]; do
+          sleep 20 # seconds
+          xvfb-run -a xfreerdp /v:${{ inputs.az_vm_hostname }} /u:${{ inputs.az_vm_username }} /p:'${{ inputs.az_vm_password }}' \
+            /cert:tofu | grep -v ${{ inputs.az_vm_hostname }} || lastExitCode=$?
+          ((--maxAttemptsCount))
+        done
+
+branding:
```

Comment:
I didn’t find any icon matching this in https://github.com/ubuntu/WSL/actions/runs/3486268478/jobs/5832591830, where is it displayed? (curious)

Reply:
Yeah I was very happy chosing icons and whatnot, but aparently they don't show up here 😞. They only appear in the Github Marketplace, if you chose to publish your action there.